[#15482] Dump threads on timeout

rigazilla opened the pull request:
fixes #15482 

if mvn verify takes more than 4 hrs:
- jstack is called for all the runnning java processes
- jstack output is provided in the github log
- and in the jstack-report artifact
- the step is marked as failed

run example available at:
https://github.com/rigazilla/infinispan/actions/runs/16960454155

tristantarrant commented on .github/workflows/on_build_do_test.yml:
```diff
@@ -119,12 +119,32 @@ jobs:
+            MVN_PID=$!
+            # Wait up to 4 hours (14400 seconds)
+            SECONDS=0
+            while kill -0 $MVN_PID 2>/dev/null; do
```
Interesting, didn't know about `kill -0` as a no-op that just checks for the PID

tristantarrant commented:
In the example run, I didn't see the jstack dumps attached as artifacts. Is that expected ?

tristantarrant commented:
Can you please fix the conflicts @rigazilla

rigazilla commented:
> In the example run, I didn't see the jstack dumps attached as artifacts. Is that expected ?

they were in the surefire-test-report artifact, I moved them to a separate one (jstack-report) and updated the PR description with a new sample run
